twitchdl/playlists_auth.py

```diff
@@ -0,0 +1,103 @@
+import httpx
+from datetime import datetime
+import random
+from urllib.parse import urlparse
+from twitchdl.exceptions import ConsoleError
+from twitchdl.twitch import gql_query
+
+def fetch_twitch_data_gql(vod_id):
+    query = f'query {{ video(id: "{vod_id}") {{ broadcastType, createdAt, seekPreviewsURL, owner {{ login }} }} }}'
+    return gql_query(query)
+
+def create_serving_id():
+    chars = "0123456789abcdefghijklmnopqrstuvwxyz"
+    return ''.join(random.choice(chars) for _ in range(32))
+
+def is_valid_quality(url):
+    with httpx.Client() as client:
+        response = client.get(url)
+        if response.status_code == 200:
+            data = response.text
+            if ".ts" in data:
+                return {"codec": "avc1.4D001E"}
+            elif ".mp4" in data:
+                mp4_url = url.replace("index-dvr.m3u8", "init-0.mp4")
+                mp4_response = client.get(mp4_url)
+                if mp4_response.status_code == 200:
+                    content = mp4_response.text
+                    return {"codec": "hev1.1.6.L93.B0" if "hev1" in content else "avc1.4D001E"}
+                return {"codec": "hev1.1.6.L93.B0"}
+    return None
+
+def fetch_auth_playlist(vod_id):
+    data = fetch_twitch_data_gql(vod_id)
+
+    if not data:
+        raise ConsoleError('Invalid vodId')
+
+    vod_data = data['data']['video']
+    channel_data = vod_data['owner']
+
+    resolutions = {
+        "audio_only": {"res": "audio_only"},
+        "160p30": {"res": "284x160", "fps": 30},
+        "360p30": {"res": "640x360", "fps": 30},
+        "480p30": {"res": "854x480", "fps": 30},
+        "720p60": {"res": "1280x720", "fps": 60},
+        "1080p60": {"res": "1920x1080", "fps": 60},
+        "chunked": {"res": "1920x1080", "fps": 60}
+    }
+
+    sorted_keys = sorted(resolutions.keys(), reverse=True)
+    ordered_resolutions = {key: resolutions[key] for key in sorted_keys}
+
+    current_url = urlparse(vod_data['seekPreviewsURL'])
+    domain = current_url.hostname
+    paths = current_url.path.split('/')
+    vod_special_id = paths[paths.index(next(p for p in paths if "storyboards" in p)) - 1]
+
+    fake_playlist = f'''#EXTM3U
+#EXT-X-TWITCH-INFO:ORIGIN="s3",B="false",REGION="EU",USER-IP="127.0.0.1",SERVING-ID="{create_serving_id()}",CLUSTER="cloudfront_vod",USER-COUNTRY="BE",MANIFEST-CLUSTER="cloudfront_vod"'''
+
+    now = datetime.strptime("2023-02-10", "%Y-%m-%d")
```

Comment:
Is this a bug? Do we want the date in the past for some reason here or should we use `datetime.now()` instead?

Reply:
i am not sure, why this date was taken in the origin link, but i am guessing there are api changes involved.

As nowadays the chance of fetching old videos before "2023" should be almost be possible, maybe we can strip the allow condition where its needed.